examples/dbscan/dbscan.cpp

```diff
@@ -297,6 +297,46 @@ void printClusterSizesAndCenters(ExecutionSpace const &exec_space,
   }
 }
 
+namespace ArborX
+{
+namespace DBSCAN
+{
+// This function is required for Boost program_options to be able to use the
+// Implementation enum.
+std::istream &operator>>(std::istream &in, Implementation &implementation)
+{
+  std::string impl_string;
+  in >> impl_string;
+
+  if (impl_string == "fdbscan")
+    implementation = ArborX::DBSCAN::Implementation::FDBSCAN;
+  else if (impl_string == "fdbscan-densebox")
+    implementation = ArborX::DBSCAN::Implementation::FDBSCAN_DenseBox;
+  else
+    in.setstate(std::ios_base::failbit);
+
+  return in;
+}
+
+// This function is required for Boost program_options to use Implementation
+// enum as the default_value().
+std::ostream &operator<<(std::ostream &out,
+                         Implementation const &implementation)
```

Comment:
> This is not accurate. It is is not required by Boost. You chose to implement it because it is useful when you print out all the options.

No, you are wrong. If I comment it out, and comment out `ss << implementation;`, I get
```
<snip>/spack/linux-ubuntu18.04-broadwell/gcc-7.5.0/boost-1.75.0-n4pwspqkjegj67hodwz67al53aouudvz/include/boost/lexical_cast/detail/converter_lexical.hpp:210:13: error: static assertion failed: Source type is neither std::ostream`able nor std::wostream`able
             BOOST_STATIC_ASSERT_MSG((result_t::value || boost::has_left_shift< std::basic_ostream< type >, T >::value),
```
with the function stack pointing to
```
        ( "impl", bpo::value<Implementation>(&implementation)->default_value(Implementation::FDBSCAN), R"(implementation ("fdbscan" or "fdbscan-densebox"))")
```
So the comment is exactly correct.